### ISSUE TITLE
defaultFileChmod does not affect localconfig.php

### ISSUE THREAD

**Issue by falkoschumann** [contao/core#8178]
I set `$GLOBALS['TL_CONFIG']['defaultFileChmod'] = 0664;`. An upload with file manager set the rights to 664 but save settings in backend result to a `lcoalconfig.php`with 644.

Background: I have a shared webspace. Apache user and ftp user are different but in same group. So if Contao (Apache) set rights to 644 I can not edit the file with ftp.

I use Contao 3.5.6.

**Comment by leofeyer** [comment]
@falkoschumann Can you please try to add the following in `system/library/Contao/Config.php` in line 260:

```php
$this->Files->chmod('system/tmp/' . $strTemp, \Config::get('defaultFileChmod'));
```

So the whole paragraph looks like this:

```php
// Then move the file to its final destination
$this->Files->chmod('system/tmp/' . $strTemp, \Config::get('defaultFileChmod'));
$this->Files->rename('system/tmp/' . $strTemp, 'system/config/localconfig.php');
```

@contao/developers /cc

**Comment by aschempp** [comment]
Be aware that https://github.com/contao/core/pull/7764 might make this obsolete. Not sure if we should always re-create the file.

**Comment by falkoschumann** [comment]
I will test it tomorrow.

**Comment by falkoschumann** [comment]
@leofeyer I changed `system/modules/core/library/Contao/Config.php`, as you describe it. It works.

**Comment by leofeyer** [comment]
Fixed in 8b765a5f1acac958b72ff638f915a5d0a360d12f then.
